### ISSUE TITLE
BenchDNN failures for inner product with int8 datatypes an sum postop, when ref. impl. is used.

### ISSUE THREAD

**Issue by nSircombe** [oneapi-src/oneDNN#1649]
# Summary
A number of `benchdnn` tests for inner product fail when the ref. implementation is used.

# Version
v3.1.0 (commit 6a35e282807c1455b194265ebeb59faa95546c4a)
The failure is present in the `master` branch, but not `rls`. It appears to have been introduced with this commit: https://github.com/oneapi-src/oneDNN/commit/77ea6245076b5c1f73ea7375e3384f9cc2b247d5

# Environment
* CPU make and model
Originally observed on AArch64 (Neoverse-N1) and on x86.
* OS version (`uname -a`)
Ubuntu 20.04
* Compiler version (`gcc --version`)
GCC 11.2
* CMake version (`cmake --version`)
3.20.1

# Steps to reproduce
Build...
```
cmake -DDNNL_CPU_RUNTIME=OMP -DDNNL_BUILD_FOR_CI=ON -DDNNL_AARCH64_USE_ACL=ON ../. && make -j
```
...and run `test_ip_int8` batches with BenchDNN

```
 ./tests/benchdnn/benchdnn --ip --batch=tests/benchdnn/inputs/ip/test_ip_int8
```

# Observed behavior
A number of int8 cases with sum fail, returning `0` unexpectedly, for example...
```
> DNNL_VERBOSE=1 ./tests/benchdnn/benchdnn --ip --dir=FWD_D --cfg=u8s8u8 --attr-scales="src:common:0.25*+dst:common:2.25*+wei:per_oc:0.5*" --attr-post-ops=sum:1.5:0:s8 mb1000ic1oc16

onednn_verbose,info,oneDNN v3.1.0 (commit 6a35e282807c1455b194265ebeb59faa95546c4a)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:8
onednn_verbose,info,cpu,isa:Intel AVX2
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:a:f0 dst_f32::blocked:a:f0,,,1,26.1501
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:a:f0 dst_f32::blocked:a:f0,,,16,11.9128
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_u8::blocked:ab:f0,,,1000x16,7.0769
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:a:f0 dst_f32::blocked:a:f0,,,1,0.111084
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_s8::blocked:ab:f0,,,16x1,0.00488281
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:ab:f0 dst_u8::blocked:ab:f0,,,1000x1,0.00415039
onednn_verbose,exec,cpu,inner_product,ref:any,forward_training,src_u8::blocked:ab:f0 wei_s8::blocked:ab:f0 bia_undef::undef:: dst_u8::blocked:ab:f0,attr-scales:src0:0+dst:0+wei:1 attr-post-ops:sum:1.5:0:s8 ,,mb1000ic1oc16,3.98999
onednn_verbose,exec,cpu,gemm_api,,undef,src_f32::blocked:ab:f0 wei_f32::blocked:ba:f0 dst_f32::blocked:ab:f0,,,1000x1:1x16,84.5039
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_u8::blocked:ab:f0 dst_f32::blocked:ab:f0,,,1000x16,21.7969
[   8][DST][0:8] exp_f32:     101.333 exp:         101 got:           0 diff:     101 rdiff:       1
[  10][DST][0:10] exp_f32:     169.333 exp:         169 got:           0 diff:     169 rdiff:       1
[  11][DST][0:11] exp_f32:         118 exp:         118 got:           0 diff:     118 rdiff:       1
[  19][DST][1:3] exp_f32:     132.056 exp:         132 got:           0 diff:     132 rdiff:       1
[  29][DST][1:13] exp_f32:     130.806 exp:         131 got:           0 diff:     131 rdiff:       1
[  38][DST][2:6] exp_f32:     93.3333 exp:          93 got:           0 diff:      93 rdiff:       1
[  40][DST][2:8] exp_f32:     162.667 exp:         163 got:           0 diff:     163 rdiff:       1
[  41][DST][2:9] exp_f32:     109.111 exp:         109 got:           0 diff:     109 rdiff:       1
[  49][DST][3:1] exp_f32:     123.833 exp:         124 got:           0 diff:     124 rdiff:       1
0:FAILED (errors:2802 total:16000) __REPRO: --ip --dir=FWD_D --cfg=u8s8u8 --attr-scales=src:common:0.25*+dst:common:2.25*+wei:per_oc:0.5* --attr-post-ops=sum:1.5:0:s8 mb1000ic1oc16
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total compute_ref: sum(s):0.10
```

# Expected behavior
All cases in this batch should pass, as they were prior to https://github.com/oneapi-src/oneDNN/commit/77ea6245076b5c1f73ea7375e3384f9cc2b247d5

**Comment by vpirogov** [comment]
+@dzarukin

**Comment by vpirogov** [comment]
This should be addressed by changes in input data generation: cd4ff0dbe9bc49129f5cb1b8cfe9b6771ba44873.

**Comment by nSircombe** [comment]
Yep, that's fixed it for me!

Thanks.
